opensearch-project/OpenSearch-Dashboards
# [Workspace] Enable workspace params in saved objects management page

### Description



### Issues Resolved

closes #6028 

## Screenshot


When enter saved objects management page in workspace foo. The page shows nothing as no objects have been created in this workspace.
![image](https://github.com/opensearch-project/OpenSearch-Dashboards/assets/13493605/ebd41aca-2d84-4bea-806a-cc67ef090d2d)

After importing with the ndjson file, objects can be found only in workspace foo.
![image](https://github.com/opensearch-project/OpenSearch-Dashboards/assets/13493605/1cbfadbd-fcbc-44bb-a6d9-8af77b256cdd)

![image](https://github.com/opensearch-project/OpenSearch-Dashboards/assets/13493605/effba2ef-42c0-4032-9158-85008a733901)

By switching to another workspace bar, no objects can be found because the objects imported before was in workspace foo
![image](https://github.com/opensearch-project/OpenSearch-Dashboards/assets/13493605/15398008-51dc-481c-9529-73c009fd87c0)


## Testing the changes



- Using the branch to bootstrap
- Download the patch file [test.patch](https://github.com/opensearch-project/OpenSearch-Dashboards/files/14535728/test.patch)
- `git apply ./test.patch`, it will
    - enable workspace feature flag
    - switch to workspace foo manually as there is not a UI component to switch workspace for now
- Start OSD by using `yarn start --no-base-path` to make sure no random base path will be appended
- Navigate to devTools
- Insert test workspaces by calling
```
PUT .kibana/_doc/workspace:foo
{
  "type": "workspace",
  "workspace": {
    "name": "foo"
  }
}

PUT .kibana/_doc/workspace:bar
{
  "type": "workspace",
  "workspace": {
    "name": "bar"
  }
}
```
- Visit the saved objects management page: `http://localhost:5601/app/management/opensearch-dashboards/objects`
- You will find nothing can be found in this workspace(After applying the patch file, you are in workspace foo)
- Import by using the [file](https://github.com/opensearch-project/OpenSearch-Dashboards/files/14535815/export.28.ndjson.zip), download the file and extract that to find the import file.
- You will find some dashboards and visualizations are imported in the workspace.
- Go to file `src/plugins/workspace/public/plugin.ts`, change the 'foo' workspace to 'bar'. Refresh the page. By doing so you are in workspace bar.
- You will find the objects can not be found again.



### Check List
